orchestra/model/remote_cache.py

```diff
@@ -47,16 +47,30 @@ def get_branches_with_update(component):
             for remote in remotes:
                 result = ls_remote(remote)
                 if result:
-                    self._cached_remote_data[component.name] = result
+                    for branch_name, commit in result.items():
+                        self._set_entry(component.name, branch_name, commit, persist_cache=False)
                     break
 
             progress_bar.update()
 
         map_to_threadpool(get_branches_with_update, clonable_components, parallelism=parallelism)
 
+        self._persist_cache()
+
+    def _persist_cache(self):
         with open(self.cache_path, "w") as f:
             json.dump(self._cached_remote_data, f)
 
+    def _set_entry(self, component_name, branch_name, commit, persist_cache=True):
+        """This helper is meant to be used by the testsuite to poison the cache
+        :param persist_cache: if True, the on-disk cache is also updated
+        """
+        current_cached_info = self._cached_remote_data.get(component_name, {})
+        current_cached_info[branch_name] = commit
+        self._cached_remote_data[component_name] = current_cached_info
+        if persist_cache:
+            self._persist_cache()
```

Comment:
Are we sure we want to enable this kind of behavior?
Can't we act on the on-disk cache directly maybe?
It seems an hack for testing, which might lead in more situations like this in other contexts.

Reply:
This method has two behaviors to allow for two use cases:
- the default sets an entry and persists on disk and is used by tests
- the non-persisting behavior is used by `rebuild_cache` (invoked by `orc update`). The changes are not persisted immediately to allow parallel fetching of the remote info (persisting immediately would be pointless and also have race conditions).
The behavior more likely to cause foot-shootings is actually the second one IMO, but it's a behavior we need. I could change `rebuild_cache` to update the in-memory cache directly and drop `persist_cache` from `_set_entry`, making it always update the on-disk cache. Do you think it's better that way?

Reply:
If you have a boolean argument that's used at the end of the function, you probably just want the caller to call another function if he wants to.

Anyway, my comment was more concerned with this method being called directly by the test suite despite being *private*.

Reply:
I removed the parameter and turned the method public (removed the leading underscore)